### DISCUSSION TITLE
Build books (compatible with existing GitBooks!) with next-mdx-books

### DISCUSSION THREAD

**Discussion by MatthewCaseres** [mdx-js/mdx#1402] [Show and tell]
I loaded [a TypeScript book](https://www.openedtech.dev/basarat/typescript-book/docs/getting-started.md) to my site to demonstrate our GitBook compatibility.

The routes are generated in Next.js with dynamic routes using some utility functions from a package I made, https://www.npmjs.com/package/next-mdx-books.

We support: 
* Content is entirely separate from source code, mdx is loaded via next-mdx-remote from a repository containing only markdown.
* Adding a new book is as simple as adding a URL to a configuration.
* load files with or without h2 tags in the side navigation (compatibility with BOTH legacy and modern style GitBooks)
*  loading files locally (for developtment) or from raw.githubusercontent.com
* provide links to source code on GitHub from each page

Plans to build a library of educational components and use overridden code blocks with YAML configurations within.

**Comment by ChristianMurphy** [comment]
Thanks for sharing!
